rust/gitxetcore/src/command/clone.rs

```diff
@@ -34,7 +32,24 @@ pub async fn clone_command(config: XetConfig, args: &CloneArgs) -> Result<()> {
     verify_user_config(None)?;
     eprintln!("Preparing to clone Xet repository.");
 
-    let (repo, _) = GitRepo::clone(
+    if args.lazy {
+        // We cannot reliably detect the directory cloned into,
+        // because of how git picks the "humanish" part of the source
+        // repository (https://git-scm.com/docs/git-clone#Documentation/git-clone.txt-ltdirectorygt),
+        // and scenarios when a directory is specified in command line arguments.
+        //
+        // So we set a env var and delegate setting the lazy configuration
+        // to the filter command.
+
+        eprintln!(
```

Comment:
Let's also log this (thinking of the scenario down the road where we are only able to use user's logs to debug things).